app/routes/PairingProcess/components/CreateAccount/pairingNotifications.js

```diff
@@ -0,0 +1,62 @@
+import firebase from 'firebase'
+import EthUtil from 'ethereumjs-util'
+import Web3 from 'web3'
+
+import config from '../../../../../config'
+
+export const setUpNotifications = () => {
+  const messaging = setUpFirebase()
+  return messaging.getToken()
+}
+
+const setUpFirebase = () => {
+  const configFirebase = {
+    authDomain: config.firebase.authDomain,
+    databaseURL: config.firebase.databaseURL,
+    projectId: config.firebase.projectId,
+    storageBucket: config.firebase.storageBucket,
+    messagingSenderId: config.firebase.messagingSenderId
+  }
+  firebase.initializeApp(configFirebase)
+  return firebase.messaging()
+}
+
+export const authPushNotificationService = (pushToken, privateKey) => {
+  const url = config.pushNotificationServiceUrl
+  const headers = {
+    'Accept': 'application/json',
+    'Content-Type': 'application/json',
+  }
+  const body = generateAuthContent(pushToken, privateKey)
+
+  return fetch(url, {
+    method: 'POST',
+    headers,
+    body,
+  })
+    .then((response) => {
+      return (response.status === 201)
+        ? true
+        : false
+    })
+    .catch((err) => {
+      console.error(err)
+      return false
+    })
+}
+
+const generateAuthContent = (pushToken, privateKey) => {
```

Comment:
Just for my interest :), can you explain to me what is going on here?

Reply:
This message is sent to the push notification service from the extension in order to be authorized to receive notifications and to pair the extension with the mobile app.
```
{
  "pushToken": <string>,
  "signature": { // signs sha3("GNO" + <pushToken>)
    "v": <integer>,
    "r": <string>, # stringified int
    "s": <string> # stringified int
  },
}
```

Reply:
Got it. Really clever the push notification service. 

Anyway, can you leave a comment just before the function with the explanation you have just described? Probably in a couple of weeks I will forget it :)